### PR TITLE
Resolves #8626. Added quarkus:remove-extension

### LINKED ISSUE #8626: Add quarkus:remove-extension
**Description**
Every pencil needs an eraser.  If there is a `quarkus:add-extension` there must be `quarkus:remove-extension`. Example use case. Someone uses 'logging-json' and they don't like the verbosity, they can go back easy the way they came in.

**Implementation ideas**
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/remove-extension.3F

### PR DESCRIPTION
**Pull request by mgubaidullin**
Implemented remove extension with Maven and Gradle

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusRemoveExtension.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        Set<String> extensionsSet = getExtensionsToRemove()
+                .stream()
+                .flatMap(ext -> stream(ext.split(",")))
+                .map(String::trim)
+                .collect(toSet());
+        try {
+            GradleBuildFile gradleBuildFile = getProject().getParent() == null
+                    ? new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()))
+                    : new GradleBuildFile(new FileProjectWriter(getProject().getProjectDir()),
+                            new FileProjectWriter(getProject().getRootProject().getProjectDir()));
```
Comment (gastaldi) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusRemoveExtension.java:47]: We should really move this code to a single place

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java [hunk 2]
```diff
@@ -12,11 +12,7 @@
 import io.quarkus.platform.descriptor.QuarkusPlatformDescriptor;
 import java.io.Closeable;
 import java.io.IOException;
-import java.util.Collections;
-import java.util.List;
-import java.util.Map;
-import java.util.Properties;
-import java.util.TreeMap;
+import java.util.*;
```
Comment (gastaldi) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java:15]: Star imports are not used, make sure your IDE is using the Quarkus code formatter

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java [hunk 3]
```diff
@@ -17,17 +17,7 @@
...
-import org.apache.maven.model.Build;
-import org.apache.maven.model.BuildBase;
-import org.apache.maven.model.Dependency;
-import org.apache.maven.model.DependencyManagement;
-import org.apache.maven.model.Model;
-import org.apache.maven.model.Plugin;
-import org.apache.maven.model.PluginExecution;
-import org.apache.maven.model.PluginManagement;
-import org.apache.maven.model.Profile;
+import org.apache.maven.model.*;
```
Comment (gastaldi) [inline independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java:20]: See https://github.com/quarkusio/quarkus/pull/8714/files#r411784431

**Review by gastaldi — COMMENTED** [review]
Added some comments

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusRemoveExtension.java (hunk 1 above)
Comment (mgubaidullin) [reply devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusRemoveExtension.java]: Moved into a method in QuarkusPlatformTask

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java (hunk 2 above)
Comment (mgubaidullin) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/BuildFile.java]: Fixed

File: independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java (hunk 3 above)
Comment (mgubaidullin) [reply independent-projects/tools/common/src/main/java/io/quarkus/cli/commands/file/MavenBuildFile.java]: Fixed

**Comment by gastaldi** [comment]
LGTM. Can you squash your commits?

**Review by aloubyansky — APPROVED** [review]

**Comment by aloubyansky** [comment]
@mgubaidullin could you please rebase your branch? There is a conflict in the gradle plugin. Otherwise, it looks good. Thanks!

**Comment by gastaldi** [comment]
Thank you!

**Pull request merged**